grc/core/Connection.py

```diff
@@ -113,9 +147,26 @@ def export_data(self):
         Export this connection's info.
 
         Returns:
-            a nested data odict
+            A tuple with connection info, and parameters.
         """
-        return (
+        con_data = (
             self.source_block.name, self.source_port.key,
-            self.sink_block.name, self.sink_port.key
+            self.sink_block.name, self.sink_port.key,
         )
+        # We only attach params if they exist, this will help with
+        # backward-compatibility. This way, regular graphs without connection
+        # parameters won't touch the connection format.
+        if self.params:
+            con_data += (collections.OrderedDict(sorted(
```

Comment:
NIT: consider making con_data a list and append to it here

Reply:
Sure why not.